tests/integration/v2/test_integration_v2.py

```diff
@@ -16,112 +20,146 @@
-        self, ops_test: OpsTest
-    ):
-        assert ops_test.model
-        requirer_unit = ops_test.model.units[f"{TLS_CERTIFICATES_REQUIRER_APP_NAME}/0"]
-        assert requirer_unit
-
-        action = await requirer_unit.run_action(action_name="get-certificate")
-
-        action_output = await ops_test.model.get_action_output(
-            action_uuid=action.entity_id, wait=60
-        )
-        assert "ca" in action_output and action_output["ca"] is not None
-        assert "certificate" in action_output and action_output["certificate"] is not None
-        assert "chain" in action_output and action_output["chain"] is not None
-
-    async def test_given_additional_requirer_charm_deployed_when_relate_then_requirer_received_certs(  # noqa: E501
-        self, ops_test: OpsTest
-    ):
-        assert ops_test.model
-        new_requirer_app_name = "new-tls-requirer"
-        await ops_test.model.deploy(
-            TestIntegration.requirer_charm, application_name=new_requirer_app_name, series="jammy"
-        )
-        await ops_test.model.add_relation(
-            relation1=new_requirer_app_name,
-            relation2=TLS_CERTIFICATES_PROVIDER_APP_NAME,
-        )
-        await ops_test.model.wait_for_idle(
-            apps=[
-                TLS_CERTIFICATES_PROVIDER_APP_NAME,
-                new_requirer_app_name,
-            ],
-            status="active",
-            timeout=1000,
-        )
-        requirer_unit = ops_test.model.units[f"{new_requirer_app_name}/0"]
-        assert requirer_unit
-
-        action = await requirer_unit.run_action(action_name="get-certificate")
-
-        action_output = await ops_test.model.get_action_output(
-            action_uuid=action.entity_id, wait=60
-        )
-        assert "ca" in action_output and action_output["ca"] is not None
-        assert "certificate" in action_output and action_output["certificate"] is not None
-        assert "chain" in action_output and action_output["chain"] is not None
-
-    async def test_given_enough_time_passed_then_certificate_expired(self, ops_test: OpsTest):  # noqa: E501
-        assert ops_test.model
-        await ops_test.model.wait_for_idle(
-            apps=[
-                TLS_CERTIFICATES_REQUIRER_APP_NAME,
-            ],
-            status="blocked",
-            timeout=1000,
-        )
-        requirer_unit = ops_test.model.units[f"{TLS_CERTIFICATES_REQUIRER_APP_NAME}/0"]
-        assert requirer_unit
-
-        assert requirer_unit.workload_status_message == "Told you, now your certificate expired"
+def build_charm(path: Path) -> Path:
+    shutil.copyfile(src=LIB_DIR, dst=path / Path(LIB_DIR))
+    _ = subprocess.run(
+        ["charmcraft", "pack", "--verbose"],
+        check=True,
+        capture_output=True,
+        encoding="utf-8",
+        cwd=path,
+    )
+    return next(path.glob("*.charm"))
+
+
+@pytest.fixture(scope="module")
+def requirer_charm():
+    charm_path = ""
```

Comment:
Why do we initialize charm path as an empty string here and in `provider_charm`?

Reply:
Good catch, removed